prototypes/rohit_magdum/VSIDemos/VSIUtilPlus/VSIUtilPlus.cpp

```diff
@@ -0,0 +1,361 @@
+		{
+			MessageBoxA(NULL, "Failed to get to foreground.", "Error", MB_OK);
+		}
+		
+		HWND hwnd = SetFocus(mhWnd);
+		if (hwnd == NULL)
+		{
+			MessageBoxA(NULL, "Failed to set focus.", "Error", MB_OK);
+		}
+
+		VSIUtilSceneInit();
+		while (this->mbDone == false)
+		{
+			
+			if (PeekMessage(&msg, NULL, 0, 0, PM_REMOVE))
+			{
+				if (msg.message == WM_QUIT)
+					this->mbDone = true;
+				else
+				{
+					TranslateMessage(&msg);
+					DispatchMessage(&msg);
+				}
+			}
+			else
+			{
+				if (this->mbIsActiveWindow == true)
+				{
+				
+					if (this->mbIsEscapeKeyPressed == true)
+					{
+						this->mbDone = true;
+					}
+					VSIUtilRender();
+					VSIUtilUpdate();
+					SwapBuffers(mhDC);
+				}
+			}
+		}
+
+		VSIUtilSceneDeinit();
+		Uninitialize();
+
+		return msg.wParam;
+	}
+
+	void VSIUtilPlus::Initialize()
+	{
+		int iPixelFormat;
+		static PIXELFORMATDESCRIPTOR pfd;
+
+		ZeroMemory(&pfd, sizeof(PIXELFORMATDESCRIPTOR));
+
+		pfd.nSize = sizeof(PIXELFORMATDESCRIPTOR);
+		pfd.nVersion = 1;
+		pfd.dwFlags = PFD_DRAW_TO_WINDOW | PFD_SUPPORT_OPENGL | PFD_DOUBLEBUFFER;
+		pfd.iPixelType = PFD_TYPE_RGBA;
+		pfd.cColorBits = 32;
+		pfd.cRedBits = 8;
+		pfd.cGreenBits = 8;
+		pfd.cBlueBits = 8;
+		pfd.cAlphaBits = 8;
+		pfd.cDepthBits = 32;
+
+		if (!(mhDC = GetDC(mhWnd)))
+		{
+			MessageBoxA(NULL, "Can not create a GL Device Context", "ERROR", MB_OK | MB_ICONEXCLAMATION);
+			exit(0);
+		}
+
+		if (!(iPixelFormat = ChoosePixelFormat(mhDC, &pfd)))
+		{
+			MessageBoxA(NULL, "Can not find suitable pixel format", "ERROR", MB_OK | MB_ICONEXCLAMATION);
+			exit(0);
+		}
+
+		if (!SetPixelFormat(mhDC, iPixelFormat, &pfd))
+		{
+			MessageBoxA(NULL, "Can not set the pixelformat", "ERROR", MB_OK | MB_ICONEXCLAMATION);
+			exit(0);
+		}
+
+		if (!(mhRC = wglCreateContext(mhDC)))
+		{
+			MessageBoxA(NULL, "Can not create Rendring context", "ERROR", MB_OK | MB_ICONEXCLAMATION);
+			exit(0);
+		}
+
+		if (!wglMakeCurrent(mhDC, mhRC))
+		{
+			MessageBoxA(NULL, "Can not activate the GL Rendering context", "ERROR", MB_OK | MB_ICONEXCLAMATION);
+			exit(0);
+		}
+
+		GLenum err = glewInit();
+		if (err != GLEW_OK)
+		{
+			MessageBoxA(NULL, "glewInit failed.", "ERROR", MB_OK);
+			exit(0);
+		}
+
+		glEnable(GL_DEPTH_TEST);
+
+		VSIUtilResize(DEFAULT_WIDTH, DEFAULT_HEIGHT);
+	}
+
+	void VSIUtilPlus::Uninitialize()
+	{
+		if (mbFullscreen == true)
+		{
+			mdwStyle = GetWindowLong(mhWnd, GWL_STYLE);
+			SetWindowLong(mhWnd, GWL_STYLE, mdwStyle | WS_OVERLAPPEDWINDOW);
+			SetWindowPlacement(mhWnd, &mwpPrev);
+			SetWindowPos(mhWnd, HWND_TOP, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE | SWP_NOOWNERZORDER | SWP_NOZORDER | SWP_FRAMECHANGED);
+
+			ShowCursor(TRUE);
+		}
+
+		wglMakeCurrent(NULL, NULL);
+
+		if (mhRC)
+		{
+			wglDeleteContext(mhRC);
+			mhRC = NULL;
+		}
+
+		if (mhDC)
+		{
+			ReleaseDC(mhWnd, mhDC);
+			mhDC = NULL;
+		}
+	}
+
+	LRESULT CALLBACK VSIUtilPlus::mWndProc(HWND hWnd, UINT iMsg, WPARAM wParam, LPARAM lParam)
+	{
+		
```

Comment:
Can we correct formatting here in general.

It will be great if we enable "replace tabs with spaces" in visual studio text editor options.